aea/contracts/base.py

```diff
@@ -121,3 +121,62 @@ def from_config(cls, configuration: ContractConfig) -> "Contract":
         #     contract_interface = json.load(interface_file)
 
         return contract_class(configuration)
+
+    # TODO if 'ContractApiMessage' was a default package
+    #   we could import it and remove the 'type ignore' tag.
+    #   the same applies to the below methods.
```

Comment:
the idea here is that methods like `get_deploy_transaction` let the developer to manually handle contract API requests (in this case `GET_DEPLOY_TRANSACTION` requests). That would mean read the `callable` field to dispatch to the right method, or just processing the request (hence ignoring the `callable` field). The same applies to the other methods below.

Reply:
An alternative would be to have the exact parameters of a "get deploy transaction callable", that is `(api, **kwargs)`, but in that case, the developer cannot do dispatching because he has not access to the `message`.

Reply:
None of the methods should be instance methods; there is no meaningful instance. We give the illusion that we keep instances alive when we don't. So either class or static method please.

Reply:
I am in favour of the approach we already implemented for the existing contract: `ledger_api, **kwargs`. The developer should not have access to the message here. All the message content is available so the developer is not constraint at all.

Reply:
... and `contract_address` for some performatives.